webbeans-impl/src/main/java/org/apache/webbeans/proxy/InterceptorDecoratorProxyFactory.java

```diff
@@ -613,5 +615,55 @@ else if (methodIndex < 32267)
         mv.visitEnd();
     }
 
+    public static class AnnotationTypeKey<T>
+    {
+        private final AnnotatedType<T> annotatedType;
+
+
+        public AnnotationTypeKey(final AnnotatedType<T> annotatedType)
+        {
+            Asserts.assertNotNull(annotatedType, "annotatedType");
+            this.annotatedType = annotatedType;
```

Comment:
i'd cache (=precompute in the constructor) the hashcode as in beancachekey since it is hash structure oriented